code/datums/uplink_items/uplink_general.dm

```diff
@@ -126,31 +126,34 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 		if(cost > U.uses)
 			return FALSE
 
-		var/obj/I = spawn_item(get_turf(user), U)
-
-		if(I)
-			if(ishuman(user))
-				var/mob/living/carbon/human/A = user
-				if(limited_stock > 0)
-					log_game("[key_name(user)] purchased [name]. [name] was discounted to [cost].")
-					if(!user.mind.special_role)
-						message_admins("[key_name_admin(user)] purchased [name] (discounted to [cost]), as a non antagonist.")
-
-				else
-					log_game("[key_name(user)] purchased [name].")
-					if(!user.mind.special_role)
-						message_admins("[key_name_admin(user)] purchased [name], as a non antagonist.")
+		if(limited_stock > 0) // only decrement it if it's actually limited
+			limited_stock--
```

Comment:
This should be done after line 136. Else a discounted item will appear as not discounted in the logging